File: app/src/main/java/in/kashewdevelopers/mirrortext/SplashScreen.java
package in.kashewdevelopers.mirrortext;

import androidx.appcompat.app.AppCompatActivity;

import android.animation.Animator;
import android.animation.AnimatorInflater;
import android.animation.AnimatorListenerAdapter;
import android.animation.AnimatorSet;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

public class SplashScreen extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        TextView appName = findViewById(R.id.appName);

        AnimatorSet animatorSet = (AnimatorSet) AnimatorInflater
                .loadAnimator(this, R.animator.splash_screen_rotation_animation);
        animatorSet.setTarget(appName);
        animatorSet.start();

        animatorSet.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationEnd(Animator animation) {
                super.onAnimationEnd(animation);
                new Handler().postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        startActivity(new Intent(SplashScreen.this, MainActivity.class));
                    }
                }, 500);
            }
        });
    }

}

File: app/src/main/java/in/kashewdevelopers/mirrortext/SplashScreen.kt
package `in`.kashewdevelopers.mirrortext

import `in`.kashewdevelopers.mirrortext.databinding.ActivitySplashScreenBinding
import android.animation.AnimatorInflater
import android.animation.AnimatorSet
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import androidx.core.animation.addListener

class SplashScreen : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivitySplashScreenBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val animatorSet = AnimatorInflater
                .loadAnimator(this, R.animator.splash_screen_rotation_animation) as AnimatorSet
        animatorSet.setTarget(binding.appName)
        animatorSet.start()

        animatorSet.addListener({
            Handler().postDelayed(Runnable {
                startActivity(Intent(this@SplashScreen, MainActivity::class.java))
            }, 500)
        })
    }

}